src/main/java/org/mandas/docker/client/DockerHost.java

```diff
@@ -81,12 +72,17 @@ private DockerHost(final String endpoint, final String certPath) {
       this.bindUri = URI.create(endpoint);
     } else {
       final String stripped = endpoint.replaceAll(".*://", "");
-      final HostAndPort hostAndPort = HostAndPort.fromString(stripped);
-      final String hostText = hostAndPort.getHost();
-      final String scheme = isNullOrEmpty(certPath) ? "http" : "https";
-
-      this.port = hostAndPort.getPortOrDefault(defaultPort());
-      this.address = isNullOrEmpty(hostText) ? DEFAULT_ADDRESS : hostText;
+      final String scheme = certPath == null || "".equals(certPath.trim()) ? "http" : "https";
```

Comment:
It might be possible for certPath to be non-null but not exist.  Would it make sense to check if certPath.exists() and has certs in it?  I'm not sure, just through of it after comparing with the similar code [here](https://github.com/dmandalidis/docker-client/pull/78/files#diff-497f14a9cf87565d6db0dab39d4d5a47R2879)

Reply:
The `DockerHost#from(String, String)` is publicly exposed so technically the `certPath` can be anything. 

In general, I really didn't question the existing codebase to avoid mixing additional refactorings which would make this PR unreviewable. My gut feeling is that we 're checking too much stuff and we 're doing extensive checks just for protecting the clients sending us nonsense. I don't remember where, but I saw it multiple times resulting in correlation problems between what you 've passed and the error we print after sanitization. We could do the check you propose and fail early but I believe this is not in scope for this PR as well as addressing the code duplication between this method and the similar one in `DefaultDockerClient`

Reply:
Ok, sounds good, thanks for for explaining.